base/strings/basic.jl

```diff
@@ -305,6 +345,26 @@ function nextind(s::AbstractString, i::Integer)
     next(s,e)[2] # out of range
 end
 
+function nextind(s::AbstractString, i::Integer, nchar::Integer)
+    nchar > 0 || throw(ArgumentError("nchar must be greater than 0"))
+    e = endof(s)
+    j = Int(i)
+    while nchar > 0
+        if j < 1
+            j = 1
+        else
+            j > e && return j+1
+            j == e && return next(s,e)[2]
+            for j = j+1:e
+                isvalid(s,j) && break
+            end
+            isvalid(s,j) || next(s,e)[2] # out of range
```

Comment:
I don't understand why you need to call `isvalid` again here, since it has been called in the last iteration already.

Reply:
You are right. Line 361 was not needed at all as if `j==e` after the loop we are sure that `s[e]` is valid (as this is the contract of `endof`)